src/ordering.jl

```diff
@@ -0,0 +1,33 @@
+export MonomialOrdering
+export GradedLex, GradedReverseLex
+export ordering
+
+"""
+    AbstractOrdering
+
+Abstract type for a monomial ordering.
+"""
+abstract type AbstractOrdering end
+
+"""
+    GradedLex
+
+    Graded lexicographical monomial ordering.
+    Compares total monomial degree first, then breaks ties lexicographically.
+    Currently, a default option for all `AbstractPolynomialLike{T}`.
+"""
+struct GradedLex <: AbstractOrdering end
+
+"""
+    GradedReverseLex
```

Comment:
No need to add this one at the moment, we can add ordering when there is an implementation that actually implements it